Realm/Realm.Weaver/Analytics/Analytics.cs

```diff
@@ -80,126 +84,131 @@ internal Analytics(Config config, ImportedReferences references, ILogger logger)
-                },
-                ["ISet`1"] = member =>
-                {
-                    if (!(member is PropertyDefinition property))
-                    {
-                        return default;
-                    }
-
-                    var keyToSet = ((GenericInstanceType)property.PropertyType).GenericArguments[0].IsPrimitive ?
-                        "PrimitiveSet" : "ReferenceSet";
-
-                    return ((_realmFeaturesToAnalyze["PrimitiveSet"] &
-                        _realmFeaturesToAnalyze["ReferenceSet"]) == 0x1, keyToSet);
-                },
-                ["RealmInteger`1"] = member => (true, "RealmInteger"),
-                ["BacklinkAttribute"] = member => (true, "BacklinkAttribute")
+                    (true, Feature.RealmObjectReference) : default,
+                [Feature.RealmValue] = member => (true, Feature.RealmValue),
+                ["IList`1"] = member => AnalyzeCollectionProperty(member, Feature.PrimitiveList, Feature.ReferenceList),
+                ["IDictionary`2"] = member => AnalyzeCollectionProperty(member, Feature.PrimitiveDictionary, Feature.ReferenceDictionary, 1),
+                ["ISet`1"] = member => AnalyzeCollectionProperty(member, Feature.PrimitiveSet, Feature.ReferenceSet),
+                ["RealmInteger`1"] = member => (true, Feature.RealmInteger),
+                [Feature.BacklinkAttribute] = member => (true, Feature.BacklinkAttribute)
             };
 
             _apiAnalysisSetters = new ()
             {
-                ["GetInstanceAsync"] = instruction => IsInRealmNamespace(instruction.Operand) ? (true, "GetInstanceAsync") : default,
-                ["GetInstance"] = instruction => IsInRealmNamespace(instruction.Operand) ? (true, "GetInstance") : default,
-
-                // ["NOT_SUPPORTED_YET"] = instruction => (true, NOT_SUPPORTED_YET),
-                ["Find"] = instruction => IsInRealmNamespace(instruction.Operand) ? (true, "Find") : default,
-                ["WriteAsync"] = instruction => IsInRealmNamespace(instruction.Operand) ? (true, "WriteAsync") : default,
-                ["ThreadSafeReference"] = instruction => (true, "ThreadSafeReference"),
+                [Feature.GetInstanceAsync] = instruction => AnalyzeRealmApi(instruction, Feature.GetInstanceAsync),
+                [Feature.GetInstance] = instruction => AnalyzeRealmApi(instruction, Feature.GetInstance),
+                [Feature.Find] = instruction => AnalyzeRealmApi(instruction, Feature.Find),
+                [Feature.WriteAsync] = instruction => AnalyzeRealmApi(instruction, Feature.WriteAsync),
+                [Feature.ThreadSafeReference] = instruction => AnalyzeRealmApi(instruction, Feature.ThreadSafeReference),
 
                 // check if it's the right signature, that is 2 params in total of which
                 // the second a bool and that it's set to true.
-                ["Add"] = instruction =>
+                // TODO - Nikola: why?
```

Comment:
Per comment - not sure why we're looking for an `Add` that takes two params where the second is true?

Reply:
Hm... I guess we expect that everyone is doing insertions, so there's no point in tracking that?